File: src/users-java/src/main/java/com/users/domain/usecase/UserUseCase.java
package com.users.domain.usecase;

import com.users.domain.api.IUserServicePort;
import com.users.domain.model.UserModel;
import com.users.domain.spi.IUserPersistencePort;

public class UserUseCase implements IUserServicePort {

    private IUserPersistencePort userPersistencePort;

    public UserUseCase(IUserPersistencePort userPersistencePort) {
        this.userPersistencePort = userPersistencePort;
    }

    @Override
    public UserModel deleteUser(Long userId) {
        return userPersistencePort.deleteUser(userId);
    }

    @Override
    public UserModel updateUser(Long id, UserModel user) {
        return userPersistencePort.updateUser(id, user);
    }
}

File: src/users-node/domain/useCase/UserUseCase.ts
import { IUserServicePort } from "../api/IUserServicePort";
import { IUserPersistencePort } from "../spi/IUserPersistencePort";

export class UserUseCase implements IUserServicePort {
  constructor(private userPersistencePort: IUserPersistencePort) {}

  async getUsers(): Promise<User[]> {
    return await this.userPersistencePort.getUsers();
  }

  async saveUser(user: User): Promise<User> {
    return await this.userPersistencePort.saveUser(user);
  }
}
